### PR TITLE
[SPARK-9335][Tests]Enable Kinesis tests only when files in extras/kinesis-asl are changed

### PR DESCRIPTION
**Pull request by zsxwing**

### CODE REVIEW

**Comment by zsxwing** [comment]
cc @tdas

**Comment by zsxwing** [comment]
@JoshRosen could you take a look?

File: dev/sparktestsupport/modules.py [hunk 1]
```diff
@@ -126,15 +129,22 @@ def contains_file(self, filename):
...
-    dependencies=[streaming],
+    dependencies=[],
     source_file_regexes=[
         "extras/kinesis-asl/",
     ],
     build_profile_flags=[
         "-Pkinesis-asl",
     ],
+    environs={
+        "ENABLE_KINESIS_TESTS": "1"
```
Comment (zsxwing) [inline dev/sparktestsupport/modules.py]: @JoshRosen Kinesis unit tests will run when `ENABLE_KINESIS_TESTS` is `1`.

Comment (zsxwing) [inline dev/sparktestsupport/modules.py]: Now only `KinesisBackedBlockRDDSuite` is controlled by `ENABLE_KINESIS_TESTS`.

**Comment by tdas** [comment]
Jenkins, test this please

**Comment by JoshRosen** [comment]
After a cursory glance this seems reasonable to me; thanks for adding such detailed code comments to explain what's going on here.

I'm cool with merging this but let's make sure that we add a "test Kinesis" item to our release checklists so that we run this test right before packaging the release candidates.

File: dev/sparktestsupport/modules.py [hunk 2]
```diff
@@ -29,7 +29,7 @@ class Module(object):
     changed.
     """
 
-    def __init__(self, name, dependencies, source_file_regexes, build_profile_flags=(),
+    def __init__(self, name, dependencies, source_file_regexes, build_profile_flags=(), environs={},
```
Comment (JoshRosen) [inline dev/sparktestsupport/modules.py]: Minor naming nit: I think `environ` would be _slightly_ more idiomatic, given that Python calls the dictionary itself `os.environ`.

**Comment by zsxwing** [comment]
> I'm cool with merging this but let's make sure that we add a "test Kinesis" item to our release checklists so that we run this test right before packaging the release candidates.

@JoshRosen where should I add the `test Kinesis` item?

**Comment by tdas** [comment]
LGTM from my side, but I will wait for @JoshRosen to LGTM it since he is most knowledgeable about this.

**Comment by tdas** [comment]
I am merging this. Spoke to @JoshRosen offline and he is cool with it. Thanks!

**Pull request closed without merging**